src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -156,31 +176,61 @@ private static void writeReadsSingle(
         }).mapToPair(t -> t);
 
         deleteHadoopFile(outputFile);
-        finalOut.saveAsNewAPIHadoopFile(outputFile, GATKRead.class, SAMRecordWritable.class, SparkBAMOutputFormat.class);
-        renameHadoopSingleShard(outputFile);
+        // Use SparkHeaderlessBAMOutputFormat so that the header is not written for each part file
+        finalOut.saveAsNewAPIHadoopFile(outputFile, GATKRead.class, SAMRecordWritable.class, SparkHeaderlessBAMOutputFormat.class);
+        mergeBam(outputFile, header);
+    }
+
+    public static <K, V> JavaPairRDD<K, V> totalSortByKey(JavaPairRDD<K, V> rdd, Comparator<K> comparator, Class<K> keyClass) {
```

Comment:
From @davidadamsphd (github lost this comment during rebase):

"Please add a comment here explain what the function is doing (and what a RangePartitioner is, since we don't use them all over our codebase)."